### PR TITLE
Use proper Sybil parsers

### PR DESCRIPTION
**Pull request by hynek**
Continuing from #2

So @cjw296 I think this is what you had in mind? It doesn't seem to work and `%` also isn't a Markdown thing, but a MyST thing if I'm not wrong (I'm trying to sound constructively helpful, not entitled, sorry if I'm failing I'm still traveling).

But I guess I can wrap the % thing in `<!-- -->` tho if we can make it work.

### CODE REVIEW

**Comment by cjw296** [comment]
MD reminds me of MySQL in that respect: a billion different slightly incompatible forks ;-) 
Does GFM have a way of specifying single-line comments?
[This change](https://github.com/simplistix/sybil/commit/101d428377eed2afaa93749be99d5036f7b07879) was requested, does it just need to be a config option?

File: conftest.py [hunk 1]
```diff
@@ -7,18 +7,29 @@
...
 import svc_reg
 
 
-pytest_collect_file = Sybil(
+markdown_examples = Sybil(
     parsers=[
-        DocTestParser(optionflags=ELLIPSIS),
-        PythonCodeBlockParser(),
+        myst.DocTestDirectiveParser(optionflags=ELLIPSIS),
+        myst.PythonCodeBlockParser(),
```
Comment (cjw296) [inline conftest.py:18]:
```suggestion
        myst.PythonCodeBlockParser(doctest_optionflags=ELLIPSIS),
```

The `PythonCodeBlockParser` also handles the doctests inside fenced code blocks.

File: README.md [hunk 2]
```diff
@@ -50,7 +50,11 @@ The latter already works with [Flask](#flask).
 
 You set it up like this:
 
+% skip: next
```
Comment (cjw296) [inline README.md:53]: Maybe just `% skip: start` to cut out the noise and move it down as you get things working?

Comment (hynek) [reply README.md:53]: I tried that but it didn't work either.

**Comment by hynek** [comment]
> Does GFM have a way of specifying single-line comments?
> [This change](https://github.com/simplistix/sybil/commit/101d428377eed2afaa93749be99d5036f7b07879) was requested, does it just need to be a config option?

I believe everyone just uses HTML comments: https://github.github.com/gfm/#example-648 – so the answer is no.

File: README.md (hunk 2 above)
Comment (cjw296) [reply README.md:66]: Did you remove the `% skip: next`? I found a bug where a `%skip: start` followed by a `% skip: next` disables the `% skip: start`. I think that nesting should raise an exception so working on that.

**Comment by cjw296** [comment]
>> Does GFM have a way of specifying single-line comments?
[This change](https://github.com/simplistix/sybil/commit/101d428377eed2afaa93749be99d5036f7b07879) was requested, does it just need to be a config option?

> I believe everyone just uses HTML comments: https://github.github.com/gfm/#example-648 – so the answer is no.

Yay! Consistency! I guess skip and other single line comment things on MyST also need to be expressable as:
 `<!-- skip: next -->`?

**Comment by hynek** [comment]
> > > Does GFM have a way of specifying single-line comments?
> > > [This change](https://github.com/simplistix/sybil/commit/101d428377eed2afaa93749be99d5036f7b07879) was requested, does it just need to be a config option?
> 
> > I believe everyone just uses HTML comments: https://github.github.com/gfm/#example-648 – so the answer is no.
> 
> Yay! Consistency! I guess skip and other single line comment things on MyST also need to be expressable as: `<!-- skip: next -->`?

Yeah, this would be the best. I've solved it using:

```markdown
<!--
; skip: next
-->
```

But

```markdown
<!-- skip: next -->
```

would obviously be much nicer.

**Comment by hynek** [comment]
Big success, thank you!

**Comment by cjw296** [comment]
@hynek - so I'm chipping away at things from the Sybil side, I'm actually hoping to "officially" support GH flavoured markdown at some point, and I'd like you ask for your review on that PR if that's okay?

In the meantime, it looks like you're actually using MyST for the docs, can I just check that [this commenting pattern](https://github.com/hynek/svcs/blob/7b6913d5d52776d5228166fbda2e2dbc37385095/docs/why.md?plain=1#L58) works properly on MyST?

**Comment by hynek** [comment]
What PR do you mean? This one has been merged and Sybil doesn't have a PR that looks related.

I'm using `% skip: next` throughout and it's working.

**Comment by cjw296** [comment]
This PR ;-) 
https://github.com/simplistix/sybil/pull/99

**Pull request merged**